services/core/shared/core/compat/sdk_v5/coreCache.js

```diff
@@ -13,41 +13,27 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-const { Utils, CacheLRU } = require('lisk-service-framework');
-
-const { isEmptyArray } = Utils.Data;
-const { getAccounts } = require('./coreApi');
+const { CacheLRU } = require('lisk-service-framework');
 
 const config = require('../../../../config');
+const { knex } = require('../../../database');
 
 const cache = CacheLRU();
 const getCacheKey = (key, value) => `account:${key}:${value}`;
 
-const normalizeAccount = account => {
-	account.address = account.address.toString('hex');
-	account.token.balance = Number(account.token.balance);
-	account.sequence.nonce = Number(account.sequence.nonce);
-	account.dpos.sentVotes = account.dpos.sentVotes.map(vote => {
-		vote.delegateAddress = vote.delegateAddress.toString('hex');
-		vote.amount = Number(vote.amount);
-		return vote;
-	});
-
-	return account;
-};
-
 const getCachedAccountBy = async (key, value) => {
+	const accountsDB = await knex('accounts');
 	const cacheKey = getCacheKey(key, value);
 	let account = await cache.get(cacheKey);
 	if (!account) {
-		const result = await getAccounts({ [key]: value });
-		if (!Array.isArray(result.data) || isEmptyArray(result.data)) {
+		const result = await accountsDB.find({ [key]: value });
+		if (result === undefined) {
```

Comment:
In its current state, the implementation would always be evaluate `result === undefined` to `false`, as `accountDB.find()` would always return an array. Empty array when no result exists and with data when it exists.

```suggestion
		const [result] = await accountsDB.find({ [key]: value });
		if (result === undefined) {
```

Reply:
Done!